Runtime/Properties/TeleportProperty.cs

```diff
@@ -0,0 +1,102 @@
+    {
+        /// <inheritdoc />
+        public event EventHandler<EventArgs> Teleported;
+
+        /// <inheritdoc />
+        public bool WasUsedToTeleport { get; private set; }
+
+        private TeleportationAnchor teleportationInteractable;
+        private BoxCollider boxCollider;
+        private Renderer[] renderers;
+
+        protected override void OnEnable()
+        {
+            base.OnEnable();
+
+            WasUsedToTeleport = false;
+            
+            boxCollider = GetComponent<BoxCollider>();
+            renderers = GetComponentsInChildren<Renderer>();
+            teleportationInteractable = GetComponent<TeleportationAnchor>();
+
+            switch (teleportationInteractable.teleportTrigger)
+            {
+                case BaseTeleportationInteractable.TeleportTrigger.OnActivated:
+                    teleportationInteractable.activated.AddListener(args =>
+                    {
+                        EmitTeleported();
+                    });
+                    break;
+                case BaseTeleportationInteractable.TeleportTrigger.OnDeactivated:
+                    teleportationInteractable.deactivated.AddListener(args =>
+                    {
+                        EmitTeleported();
+                    });
+                    break;
+                case BaseTeleportationInteractable.TeleportTrigger.OnSelectEntered:
+                    teleportationInteractable.selectEntered.AddListener(args =>
+                    {
+                        EmitTeleported();
+                    });
+                    break;
+                case BaseTeleportationInteractable.TeleportTrigger.OnSelectExited:
+                    teleportationInteractable.selectExited.AddListener(args =>
+                    {
+                        EmitTeleported();
+                    });
+                    break;
+            }
+        }
+
+        /// <inheritdoc />
+        public void FastForwardTeleport()
+        {
+            TeleportRequest teleportRequest = new TeleportRequest
+            {
+                requestTime = Time.time,
+                matchOrientation = teleportationInteractable.matchOrientation,
+                destinationPosition = teleportationInteractable.teleportAnchorTransform.position,
+                destinationRotation = teleportationInteractable.teleportAnchorTransform.rotation
+            };
+
+            teleportationInteractable.teleportationProvider.QueueTeleportRequest(teleportRequest);
+        }
+
+        /// <inheritdoc />
+        protected override void InternalSetLocked(bool lockState)
+        {
+            boxCollider.enabled = !lockState;
+            teleportationInteractable.enabled = !lockState;
+
+            foreach (Renderer anchorRenderer in renderers)
+            {
+                anchorRenderer.enabled = !lockState;
+            }
+
+            if (lockState)
+            {
+                WasUsedToTeleport = false;
```

Comment:
So with locking mechanics (which are optional) this never gets reset to false?

Reply:
🤔😱